lib/lanpartyseating/logic/station_logic.ex

```
@@ -56,15 +63,18 @@ defmodule Lanpartyseating.StationLogic do
         is_closed: is_broken
       )
 
-    case Repo.update(station) do
-      {:ok, result} ->
+    update = Repo.update(station)
+
+    case update do
+      {:ok, _} ->
+        {:ok, stations} = StationLogic.get_all_stations()
         Phoenix.PubSub.broadcast(
           PubSub,
           "station_update",
-          {:stations, StationLogic.get_all_stations()}
+          {:stations, stations}
         )
-        result
-      {:error, _} -> nil
+        update
+      _ -> update
```

This should return a custom error code instead of Ecto's